workers/unity/Packages/com.improbable.gdk.core/Worker/WorkerConnector.cs

```diff
@@ -117,10 +117,12 @@ await Worker.CreateWorkerAsync(GetAlphaLocatorConfig(workerType), connectionPara
 
                 Worker.OnDisconnect += OnDisconnected;
 
-                HandleWorkerConnectionEstablished();
-
-                World.Active = World.Active ?? Worker.World;
-                ScriptBehaviourUpdateOrder.UpdatePlayerLoop(World.AllWorlds.ToArray());
+                if (Application.isPlaying)
```

Comment:
This could also be a check for not playing that returns, which is what happens elsewhere in the same file.

Reply:
However as the connection has been made by this point we will need to dispose of that to free native memory.

Reply:
will add that. 

also going to make `OnDestroy` protected as I just realized that we are not disposing the sim player coordinator correctly anymore in the other PR I made